src/MigrationTools.Clients.AzureDevops.ObjectModel/Processors/TfsPipelineProcessor.cs

```diff
@@ -0,0 +1,338 @@
+                    {
+                        targetTaskGroups.Add(taskGroup);
+                    }
+                }
+            }
+        }
+
+        private void GetBuildDefinitions(string organisation, string project, string accessToken)
+        {
+            string baseUrl = organisation + "/" + project + "/_apis/build/definitions";
+            string credentials = Convert.ToBase64String(Encoding.ASCII.GetBytes(":" + accessToken));
+
+            WebClient client = new WebClient();
+            client.Headers[HttpRequestHeader.Authorization] = "Basic " + credentials;
+            string httpResponse = client.DownloadString(baseUrl);
+
+            if (httpResponse != null)
+            {
+                BuildDefinitions pipelines = JsonConvert.DeserializeObject<BuildDefinitions>(httpResponse);
+
+                foreach (BuildDefinition pipeline in pipelines.Value)
+                {
+                    //Nessecary because getting all Pipelines doesn't include all of their properties
+                    string responseMessage = client.DownloadString(baseUrl + "/" + pipeline.Id);
+
+                    BuildDefinition newPipeline = JsonConvert.DeserializeObject<BuildDefinition>(responseMessage);
+
+                    if (organisation == Source.Organisation)
+                    {
+                        sourceBuildDefinition.Add(newPipeline);
+                    }
+                    else
+                    {
+                        targetBuildDefinition.Add(newPipeline);
+                    }
+                }
+            }
+        }
+
+        private void GetReleaseDefinitions(string organisation, string project, string accessToken)
+        {
+            string baseUrl = organisation + "/" + project + "/_apis/release/definitions";
+            string credentials = Convert.ToBase64String(Encoding.ASCII.GetBytes(":" + accessToken));
+
+            WebClient client = new WebClient();
+            client.Headers[HttpRequestHeader.Authorization] = "Basic " + credentials;
+            string httpResponse = client.DownloadString(baseUrl);
+
+            if (httpResponse != null)
+            {
+                ReleaseDefinitions pipelines = JsonConvert.DeserializeObject<ReleaseDefinitions>(httpResponse);
+
+                foreach (ReleaseDefinition pipeline in pipelines.Value)
+                {
+                    //Nessecary because getting all Pipelines doesn't include all of their properties
+                    string responseMessage = client.DownloadString(baseUrl + "/" + pipeline.Id);
+
+                    ReleaseDefinition newPipeline = JsonConvert.DeserializeObject<ReleaseDefinition>(responseMessage);
+
+                    if (organisation == Source.Organisation)
+                    {
+                        sourceReleaseDefinitions.Add(newPipeline);
+                    }
+                    else
+                    {
+                        targetReleaseDefinitions.Add(newPipeline);
```

Comment:
shouldnt this method return a list of definitions and instead the calling method knows if it is source or target?

Reply:
Yes, I changed that